Dockerfile

```diff
@@ -0,0 +1,31 @@
+FROM python:3.6.9-alpine3.10
+LABEL maintainer="Hypothes.is Project and contributors"
+
+# Install nginx & supervisor
+RUN apk add --no-cache nginx gettext supervisor
+
+# Create the hypothesis user, group, home directory and package directory.
+RUN addgroup -S hypothesis && adduser -S -G hypothesis -h /var/lib/hypothesis hypothesis
+WORKDIR /var/lib/hypothesis
+
+# Ensure nginx state and log directories writeable by unprivileged user.
+RUN chown -R hypothesis:hypothesis /etc/nginx/conf.d /var/log/nginx /var/lib/nginx /var/tmp/nginx
+
+# Copy minimal data to allow installation of python dependencies.
+COPY ./requirements.txt ./
+
+# Install build deps, build, and then clean up.
+RUN apk add --no-cache --virtual build-deps \
+  && pip install --no-cache-dir -U pip \
+  && pip install --no-cache-dir -r requirements.txt \
+  && apk del build-deps
+
+COPY ./conf/supervisord.conf ./conf/supervisord.conf
+
+COPY ./nginx/nginx.conf /etc/nginx/nginx.conf
+COPY ./nginx/nginx_envsubst.conf.template /var/lib/hypothesis/nginx_envsubst.conf.template
+COPY . .
```

Comment:
I feel like there is still some tidy up possible here. Do we need to copy everything, or just the python? Might make for a smaller image to boot.

Reply:
Agh. Idk where that copy . . came from. No we don’t need- I’ll fix that.

Reply:
We do need `Makefile`, `tox.ini` and others (linting config files, coverage config files, ...) for Jenkins (it runs the tests, formatting, linting, coverage, etc in the production Docker container).

You may question why these files are in our production Docker image when they're not used in production (only on Jenkins) but this is how all our apps do it so I'd recommend fixing it separately and for all the apps together.

Presumably the correct fix is for `Dockerfile` to _not_ copy files that're only needed for CI, and for `Jenkinsfile` to copy those files into the Docker container itself. But we don't need to hold up this PR on getting that working.

We might just be setting ourselves up for future Jenkins breakage (when we add another file that's needed) just for the sake of keeping the Docker image minimal anyway, which may be a questionable tradeoff

Reply:
It's definitely a wonky but I'm gonna go ahead and merge this as is then.